src/librustc_trans/tvec.rs

```diff
@@ -52,7 +52,7 @@ pub fn slice_for_each<'blk, 'tcx, F>(bcx: Block<'blk, 'tcx>,
     let current = Phi(header_bcx, val_ty(start), &[start], &[bcx.llbb]);
 
     let keep_going =
-        ICmp(header_bcx, llvm::IntULT, current, end, DebugLoc::None);
```

Comment:
Without the LLVM change the regression isn't fully fixed, but yeah, this is the main part of it.
It might be in different places, so searching for `IntULT` used in an iteration loop condition should help.